### ISSUE TITLE
Cover more situations when the warning about Esync-incompatible Wine should be shown

### ISSUE THREAD

**Issue by tannisroot** [lutris/lutris#1561]
Right now, if you:
1. Select an Esync-compatible Wine (tkg)
2. Activate Esync toggle
3. Switch Wine version to one that is not Esync-compatible (WineHQ ones, for example)

The esync toggle will be kept activated, despite using a Wine version that doesn't have Esync.
Instead, Lutris should check if newly selected Wine version supports Esync and if it doesn't, show the warning (and disable the option if the user doesn't check the "Enable Anyway" checkbox).

**Comment by strycore** [comment]
PR or bust

**Comment by strycore** [comment]
This involve setting signal handlers to the version widget which is a dynamically generated combo box. This is not functionality we have or I want to add. Last time we added support for such a thing, we ended up adding a hell of a mess of convoluted code. 
The error message about switching wine builds is perfectly understandable, if the user fucks up their config we're not going to make our product 10 times more complicated to have very minor and unvaluable changes in the UI like that.
